make-uniform/production/Build Standard Database.Rmd

```diff
@@ -58,265 +58,336 @@ dir_path <- user_list %>%
 
 # Prepare operator-specific dictionary
-dictionary.MUN <- dictionary.all %>%
-  filter(operator == survey.MUN.key)
+dictionary_MUN <- dictionary_all %>%
+  filter(operator == survey_MUN_key)
 
-variables.MUN <- as.character(factor(unique(unlist(dictionary.MUN$Survey_Variable, use.names = TRUE))))
+variables_MUN <- dictionary_MUN$survey_variable %>%
+  unique()
 
 # Read data and select variables of interest
-survey.MUN <- read.csv(F_MUNI_SURVEY, header = TRUE)
-for (var in variables.MUN) {
-  if (var %in% colnames(survey.MUN)==FALSE) {
-    print(paste(var, var %in% colnames(survey.MUN)))
-  }
-}
-survey.MUN.select <- survey.MUN[ ,variables.MUN]
+survey_MUN <- read.csv(f_muni_survey_path, header = TRUE)
+
+# JWH: I don't understand why this code is needed_
+# for (var in variables_MUN) {
+#   if (var %in% colnames(survey_MUN)==FALSE) {
+#     print(paste(var, var %in% colnames(survey_MUN)))
+#   }
+# }
+
+survey_MUN_select <- survey_MUN %>%
+  select(one_of(variables_MUN))
 
 # Reshape the data
-survey.MUN.melt <- melt(survey.MUN.select, id = 'id')
-survey.MUN.melt <- select(survey.MUN.melt, id, Survey_Variable = variable, Survey_Response = value)
-survey.MUN.melt <- mutate(survey.MUN.melt, operator = survey.MUN.key)
-survey.MUN.melt <- mutate(survey.MUN.melt, survey_year = survey.MUN.year)
-survey.MUN.melt <- mutate(survey.MUN.melt, survey_tech = survey.MUN.tech)
-colnames(survey.MUN.melt)[1] <- "ID"
+survey_MUN_gather <- survey_MUN_select %>%
+  gather(survey_variable, survey_response, -id) %>%
+  rename(ID = id) %>%
+  mutate(# ID = as.numeric(ID),
+         operator = survey_MUN_key,
+         survey_year = survey_MUN_year,
+         survey_tech = survey_MUN_tech
+         ) 
 
 # Clean up
-remove(survey.MUN.key, survey.MUN.year, survey.MUN.tech, dictionary.MUN, variables.MUN, survey.MUN, survey.MUN.select)
+remove(survey_MUN_key, survey_MUN_year, survey_MUN_tech, dictionary_MUN, variables_MUN, survey_MUN, survey_MUN_select)
 ```
 
 ### VTA
 ```{r operator-VTA}
-survey.VTA.key  <- 'VTA'
-survey.VTA.year <- 2017
+survey_VTA_key  <- 'VTA'
+survey_VTA_year <- 2017
 
 # Set survey tech by route if necessary in chunk `combine`
-survey.VTA.tech <- 'local bus'
+survey_VTA_tech <- 'local bus'
 
 # Prepare operator-specific dictionary
-dictionary.VTA <- dictionary.all %>%
-  filter(operator == survey.VTA.key)
+dictionary_VTA <- dictionary_all %>%
+  filter(operator == survey_VTA_key)
 
-variables.VTA <- as.character(factor(unique(unlist(dictionary.VTA$Survey_Variable, use.names = TRUE))))
+variables_VTA <- dictionary_VTA$survey_variable %>%
+  unique()
 
 # Read data and select variables of interest
-survey.VTA <- read.csv(F_VTA_SURVEY, header = TRUE)
-for (var in variables.VTA) {
-  if (var %in% colnames(survey.VTA)==FALSE) {
-    print(paste(var, var %in% colnames(survey.VTA)))
-  }
-}
-survey.VTA.select <- survey.VTA[ ,variables.VTA]
+survey_VTA <- read.csv(f_vta_survey_path, header = TRUE)
+
+# # JWH: I don't understand why this code is needed_
```

Comment:
When the data is being read in, it's, right now, a somewhat iterative and tedious process to make sure the dictionary includes all the variables you want it to include. So we should add a note here that we need some type of helper method like this when reading in a new survey. I'll create an issue for this and we can address it when we start using the code to set up new operators. For now, this code snippet can be removed.

Reply:
Issue #13